frontend/src/services/UserService.js

```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+
+const BASE_URL = "http://localhost:8080";
+
+
+class UserService {
+    
+    registerUser(user) {
+        return axios.post(BASE_URL + "/register", user);
```

Comment:
Try using async/await to assure asynchronous and promise based behaviour.

`async` will not block browser from executing other operations
`await` will pause the execution of some particular code - this is important for http request since we don't want that some block of code is executed until we get the response